build/index.js

```diff
@@ -35,124 +42,109 @@ const bootstrap = () => {
-};
+const convertIcon = (iconsPath) => async (icon) => {
+  const iconName = createIconName(basename(icon));
 
-const convertIcon = (iconsPath) => (icon) => {
-  const pathToIcon = `${iconsPath}/${icon}`;
-  const moduleName = createModuleName(basename(icon));
-  const elmModulePath = `${elmPath}/${moduleName}.elm`;
-  const convertor = require("svg-to-elm").default;
+  try {
+    const pathToIcon = `${iconsPath}/${icon}`;
+    const convertor = require("svg-to-elm").default;
+    const res = await convertor.parse(pathToIcon, { moduleName: iconName });
 
-  convertor.parse(pathToIcon, { moduleName: elmModulePath }).then((res) => {
     if (!res.success) {
       throw new Error(res.message);
     }
 
-    try {
-      const elmContent = createElmContent({
-        moduleName: `ZondIcons.${moduleName}`,
-        imports: res.imports,
-        body: res.viewBody,
-        exports: res.moduleExposing,
-      });
-
-      fs.writeFileSync(elmModulePath, elmContent);
-    } catch (e) {
-      throw new Error(`Could not write Elm (${elmModulePath}): ${e}`);
-    }
-  });
+    return createElmContent({
+      iconName,
+      body: res.viewBody,
+      funcNames: res.moduleExposing.entries || [],
+    });
+  } catch (e) {
+    throw new Error(`Could not convert ${icon}: ${e}`);
+  }
 };
+
 const isSVGIcon = (file) => path.extname(file).toLowerCase() === ".svg";
 
-const convertIcons = (iconsPath) => {
+const convertIcons = async (iconsPath) => {
   try {
     const convert = convertIcon(iconsPath);
 
-    fs.readdirSync(iconsPath)
-      .filter(isSVGIcon)
-      .forEach(convert);
-    return Promise.resolve();
+    const buffer = await Promise.all(
+      fs.readdirSync(iconsPath).reduce((acc, file) => {
+        if (isSVGIcon(file)) {
+          return [...acc, convert(file)];
+        }
+        return acc;
+      }, [])
+    );
+    const elmContent = `
+module ${elmModuleName} exposing (..)
+
+import Html exposing (Html)
+import Svg exposing (..)
+import Svg.Attributes exposing (..)
+
+${buffer.join("\n")}
+    `;
+
+    fs.writeFileSync(`${elmPath}/${elmModuleName}.elm`, elmContent);
   } catch (e) {
-    return Promise.reject(`Could not covert icons: ${e}`);
+    throw new Error(`Could not convert icons: ${e}`);
   }
 };
 
-const formatElm = () => {
+const formatElm = async () => {
   const { execSync } = require("child_process");
 
   try {
     execSync(`npx elm-format ${elmPath}/*.elm --yes`);
-    return Promise.resolve();
-  } catch (e) {
-    return Promise.reject(`Could not format Elm files: ${e}`);
-  }
-};
-
-const updateElmJson = () => {
-  try {
-    const elmJson = require(elmJsonFile);
-    elmJson["exposed-modules"] = fs
-      .readdirSync(elmPath)
-      .map((file) => `ZondIcons.${basename(file)}`);
-    fs.writeFileSync(elmJsonFile, JSON.stringify(elmJson));
-    return Promise.resolve();
   } catch (e) {
-    return Promise.reject(`Could not update elm.json: ${e}`);
+    throw new Error(`Could not format files: ${e}`);
```

Comment:
I really like it better before. Returning promises is much nicer IMO than throwing and try/catch

Reply:
Alright. I can revert those

Reply:
Done